### PR TITLE
Gather test coverage with stable rust

### PR DESCRIPTION
**Pull request by RicoHageman**
This will ensure that the test coverage can be determined using the stable rust release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Make `PYO3_CROSS_LIB_DIR` environment variable optional when cross compiling. [#2241](https://github.com/PyO3/pyo3/pull/2241)
 - Mark `METH_FASTCALL` calling convention as limited API on Python 3.10. [#2250](https://github.com/PyO3/pyo3/pull/2250)
 - Deprecate `pyo3_build_config::cross_compiling` in favour of `pyo3_build_config::cross_compiling_from_to`. [#2253](https://github.com/PyO3/pyo3/pull/2253)
+- Determine test coverage using the stable release of Rust. [#2286](https://github.com/PyO3/pyo3/pull/2286)
```
Comment (davidhewitt) [inline CHANGELOG.md:32]: TBH I don't think this needs a CHANGELOG entry; it's not really user-facing API change.

This is in the wrong place anyway; 0.16.3 has been released.

**Review by davidhewitt — APPROVED** [review]
Thanks 👍

**Comment by davidhewitt** [comment]
Not sure why the coverage jobs are failing... ?

**Comment by adamreichold** [comment]
> Not sure why the coverage jobs are failing... ?

I think our xtask binary is still forcing a nightly toolchain:

image

https://github.com/PyO3/pyo3/blob/749fe6c4bbf5ed556fbde263a2481e384f282760/xtask/src/llvm_cov.rs#L92-L97

**Review by adamreichold — APPROVED** [review]

**Pull request merged**